apps/smtp/src/pages/api/webhooks/gift-card-sent.ts

```diff
@@ -103,14 +105,44 @@ const handler: NextWebhookApiHandler<GiftCardSentWebhookPayloadFragment> = async
 
   const useCase = useCaseFactory.createFromAuthData(authData);
 
-  await useCase.sendEventMessages({
-    channelSlug: channel,
-    event: "GIFT_CARD_SENT",
-    payload,
-    recipientEmail,
-  });
-
-  return res.status(200).json({ message: "The event has been handled" });
+  return useCase
+    .sendEventMessages({
+      channelSlug: channel,
+      event: "GIFT_CARD_SENT",
+      payload,
+      recipientEmail,
+    })
+    .then((result) =>
+      result.match(
+        (r) => {
+          logger.info("Successfully sent email(s)");
+
+          return res.status(200).json({ message: "The event has been handled" });
+        },
+        (err) => {
+          const errorInstance = err[0];
+
+          if (errorInstance instanceof SendEventMessagesUseCase.ServerError) {
```

Comment:
**Question:** can't you use here switch with error constructor?

Reply:
Not really due to prototype chain. Here is demo:

![CleanShot 2024-05-20 at 09 48 58](https://github.com/saleor/apps/assets/9268745/3f66e6ff-5827-430b-ae88-5222698dff3d)